src/lib/builders/pin-input/create.ts

```diff
@@ -95,7 +96,7 @@ export function createPinInput(props?: CreatePinInputProps) {
 					placeholder: $placeholder,
 					disabled: disabledAttr($disabled),
 					type: $type,
-					value: currValue,
+					value: options.mask && currValue ? '*' : currValue,
```

Comment:
Does the hiddenInput value still reflect the correct value with this?

Reply:
Hmm I see what you mean, I was too quick on this one 😅. The hidden input receives its value from 'value' which will be the maskvalue and not the real value.

I'm on mobile so could only look briefly. The easiest way would be to set the input type to password if masked is selected, but not sure if that's compatible with letting the user change the character.